go/vt/vtgate/executor.go

```diff
@@ -1502,6 +1497,88 @@ func buildVarCharRow(values ...string) []sqltypes.Value {
 	return row
 }
 
+func generateCharsetRows(showFilter *sqlparser.ShowFilter, colNames []string) [][]sqltypes.Value {
+	if showFilter == nil {
+		return buildCharsetRows("both")
+	}
+
+	var filteredColName string
+	if showFilter.Like != "" {
+		filteredColName = checkLikeOpt(showFilter.Like, colNames)
+	}
+
+	cmpExp, ok := showFilter.Filter.(*sqlparser.ComparisonExpr)
+	if ok {
+		left, ok := cmpExp.Left.(*sqlparser.ColName)
+		if !ok {
+			panic("expect left side to be 'charset'")
```

Comment:
Since `handleShow` can return an error, maybe you can pass through the errors instead of panicking?

Reply:
thank you for pointing this out. as a general rule, we should _never_ panic in vtgate (query execution).